### PR TITLE
Add CRI-O SELinux configuration

### PR DESCRIPTION
**Pull request by jkroepke**
Fixes https://github.com/openshift/openshift-ansible/issues/10283

### CODE REVIEW

File: roles/container_runtime/defaults/main.yml [hunk 1]
```diff
@@ -80,3 +82,5 @@ l_required_docker_version: '1.13'
 l_insecure_crio_registries: "{{ '\"{}\"'.format('\", \"'.join(l2_docker_insecure_registries)) }}"
 l_crio_registries: "{{ l2_docker_additional_registries + ['docker.io'] }}"
 l_additional_crio_registries: "{{ '\"{}\"'.format('\", \"'.join(l_crio_registries)) }}"
+
+l_crio_selinux_enabled: "{{ openshift_crio_selinux_enabled | to_json }}"
```
Comment (vrutkovs) [inline roles/container_runtime/defaults/main.yml]: I don't think an additional var is required

File: roles/container_runtime/templates/crio.conf.j2 [hunk 2]
```diff
@@ -86,7 +86,7 @@ conmon_env = [
 # selinux indicates whether or not SELinux will be used for pod
 # separation on the host. If you enable this flag, SELinux must be running
 # on the host.
-selinux = true
+selinux = {{ l_crio_selinux_enabled | lower }}
```
Comment (vrutkovs) [inline roles/container_runtime/templates/crio.conf.j2]: `| bool | string` instead?

File: roles/container_runtime/defaults/main.yml (hunk 1 above)
Comment (jkroepke) [reply roles/container_runtime/defaults/main.yml]: What the pattern? I just copy it the from `openshift_docker_selinux_enabled`

Comment (vrutkovs) [reply roles/container_runtime/defaults/main.yml]: `l_docker_selinux_enabled` is required to be set in `daemon.json`, so we don't need it in this case. Also, docker's selinux setting is not a very good pattern code-wise

**Comment by sdodson** [comment]
@openshift/sig-container-security 
@openshift/sig-containers 
Is CRI-O without selinux something we intend to support?

**Comment by runcom** [comment]
> Is CRI-O without selinux something we intend to support?

I highly doubt so, @rhatdan @mrunalp ?

**Comment by sdodson** [comment]
@jeremyeder may have feedback on the need as well.

**Comment by rhatdan** [comment]
Yes it should be configurable.  We should not force people to use SELinux confinement.  BUT this really only enforces SELinux confinement between containers.  If a customer does not want to use SELinux they can disable it on the node, then this flag will have no effect.

Having this flag in docker or in cri-o really causes people to have false assumptions about whether or not SELinux is enforced.  It is enforced, but we run the containers with an unconfined label, so they can attack each other from an SELinux point of view.

**Comment by sdodson** [comment]
@rhatdan Are you then suggesting that disabling selinux at the kernel level is the correct (and sufficient) approach to disabling selinux? Or would customers be interested in selinux at host level but disabled at container runtime level?

**Comment by jeremyeder** [comment]
The use-cases I am aware of for disabling SELinux are performance-related, and thus it is disabled at the host-level.  I believe this knob is to handle the host=enabled runtime=disabled case.

What is CRI-O behavior on a node that is booted with selinux=disabled?  We want to have clusters where selinux=disabled only on certain worker nodes, and enabled everywhere else.

**Comment by rhatdan** [comment]
@sdodson I would never recommend disabling SELinux on the host, and definitely not in container separation in the container engine (CRI-O, Docker).  As @jeremyeder points out the only conceivable reasons to disable SELinux is for performance, and this should be done on the host.  Or because you specific container can not run, and this can be done in the specific containers configuration.  IE Running the container with --security-opt label:disable.  Having the Runtime have SELinux disabled, I can not think of a good reason to do this, other then  you're configuration of the container engine and its data is screwed up/mislabeled causing all containers to fail to run with SELinux, in which case you should fix the setup rather then just disable the enforcement.

**Comment by sdodson** [comment]
Thanks for the submission, but the opinion seems to be that if you wish to disable selinux it should not be disabled via the container runtime but either at a host or per container level.
/close

**Pull request closed without merging**